x/trigger/types/genesis_test.go

```diff
@@ -0,0 +1,276 @@
+			},
+			modify: func(gs *GenesisState) {
+				gs.Triggers[0].Owner = "cosmos1w6t0l7z0yerj49ehnqwqaayxqpe3u7e23edgma"
+				gs.Triggers[0].Event = badRequest.Event
+			},
+			err: "empty event name",
+		},
+		{
+			name: "invalid - A queued trigger's event must pass validation",
+			state: &GenesisState{
+				TriggerId:      2,
+				QueueStart:     1,
+				GasLimits:      []GasLimit{{TriggerId: 1, Amount: 1}, {TriggerId: 2, Amount: 1}},
+				Triggers:       []Trigger{trigger},
+				QueuedTriggers: []QueuedTrigger{{BlockHeight: 1, Time: time.Time{}, Trigger: trigger2}},
+			},
+			modify: func(gs *GenesisState) {
+				gs.Triggers[0].Owner = "cosmos1w6t0l7z0yerj49ehnqwqaayxqpe3u7e23edgma"
+				gs.QueuedTriggers[0].Trigger.Event = badRequest.Event
+			},
+			err: "empty event name",
+		},
+		{
+			name: "invalid - Gas limits must match either a trigger or queued trigger",
+			state: &GenesisState{
+				TriggerId:      3,
+				QueueStart:     1,
+				GasLimits:      []GasLimit{{TriggerId: 3, Amount: 1}, {TriggerId: 2, Amount: 1}},
+				Triggers:       []Trigger{trigger},
+				QueuedTriggers: []QueuedTrigger{{BlockHeight: 1, Time: time.Time{}, Trigger: trigger2}},
+			},
+			modify: nil,
+			err:    "trigger or queued trigger does not have a gas limit that matches it with id 1",
+		},
+		{
+			name: "invalid - Gas limits ids must be unique",
+			state: &GenesisState{
+				TriggerId:      1,
+				QueueStart:     1,
+				GasLimits:      []GasLimit{{TriggerId: 1, Amount: 1}, {TriggerId: 1, Amount: 1}},
+				Triggers:       []Trigger{trigger},
+				QueuedTriggers: []QueuedTrigger{{BlockHeight: 1, Time: time.Time{}, Trigger: trigger}},
+			},
+			modify: nil,
+			err:    "cannot have duplicate trigger id in gas limits",
+		},
+		{
+			name: "invalid - triggers cannot have duplicate id",
+			state: &GenesisState{
+				TriggerId:      1,
+				QueueStart:     1,
+				GasLimits:      []GasLimit{{TriggerId: 1, Amount: 1}, {TriggerId: 2, Amount: 1}, {TriggerId: 3, Amount: 1}},
+				Triggers:       []Trigger{trigger, trigger},
+				QueuedTriggers: []QueuedTrigger{{BlockHeight: 1, Time: time.Time{}, Trigger: trigger2}},
+			},
+			modify: nil,
+			err:    "all trigger ids shared between triggers and queued triggers must be unique",
+		},
+		{
+			name: "invalid - queued triggers cannot have duplicate id",
+			state: &GenesisState{
+				TriggerId:      2,
+				QueueStart:     1,
+				GasLimits:      []GasLimit{{TriggerId: 1, Amount: 1}, {TriggerId: 2, Amount: 1}, {TriggerId: 3, Amount: 1}},
+				Triggers:       []Trigger{trigger},
+				QueuedTriggers: []QueuedTrigger{{BlockHeight: 1, Time: time.Time{}, Trigger: trigger2}, {BlockHeight: 1, Time: time.Time{}, Trigger: trigger2}},
+			},
+			modify: nil,
+			err:    "all trigger ids shared between triggers and queued triggers must be unique",
+		},
+		{
+			name: "invalid - cannot have duplicate id between triggers and queued triggers",
+			state: &GenesisState{
```

Comment:
Thank you. Thank you. Thank you for using struct initialization directly for these test case definitions instead of `NewGenesisState`.

With this struct initialization, I know exactly what each value is. I don't need to keep looking up `NewGenesisState` to figure out which number goes to which field. Even more important when several values are just `nil`.

This makes me happy.